Allow xenstored_t manage xend_var_lib_t files (bsc#1228540)

ca-hu opened the pull request:
init-xenstore-domain adds a xenstore domain and needs to write /var/lib/xen/userdata.*.libxl-json and manage the respective domain-userdata-lock files, which are labelled xend_var_lib_t atm.

Fixes:
type=AVC msg=audit(1730402738.246:35): avc:  denied  { create } for  pid=1419 comm="init-xenstore-d" name="userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1 type=AVC msg=audit(1730402738.253:36): avc:  denied  { read write open } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1 type=AVC msg=audit(1730402738.253:37): avc:  denied  { lock } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1 type=AVC msg=audit(1730402738.253:38): avc:  denied  { getattr } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1 type=AVC msg=audit(1730402738.253:39): avc:  denied  { rename } for  pid=1419 comm="init-xenstore-d" name="userdata-n.1.00000000-0000-0000-0000-000000000000.libxl-json" dev="sdb2" ino=4719263 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1 type=AVC msg=audit(1730402738.253:40): avc:  denied  { unlink } for  pid=1419 comm="init-xenstore-d" name="userdata-d.1.00000000-0000-0000-0000-000000000000.libxl-json" dev="sdb2" ino=4719273 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1

zpytela commented:
@ca-hu We usually do not use interfaces from the same module, is there any need for that?

ca-hu commented:
thanks for checking, no not directly. i can add it to the main module as well

ca-hu committed:
Allow xenstored_t manage xend_var_lib_t files (bsc#1228540)

init-xenstore-domain adds a xenstore domain and needs to write
/var/lib/xen/userdata.*.libxl-json and manage the respective
domain-userdata-lock files, which are labelled xend_var_lib_t
atm.

Fixes:
type=AVC msg=audit(1730402738.246:35): avc:  denied  { create } for  pid=1419 comm="init-xenstore-d" name="userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1
type=AVC msg=audit(1730402738.253:36): avc:  denied  { read write open } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1
type=AVC msg=audit(1730402738.253:37): avc:  denied  { lock } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1
type=AVC msg=audit(1730402738.253:38): avc:  denied  { getattr } for  pid=1419 comm="init-xenstore-d" path="/var/lib/xen/userdata-l.1.00000000-0000-0000-0000-000000000000.domain-userdata-lock" dev="sdb2" ino=4719235 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1
type=AVC msg=audit(1730402738.253:39): avc:  denied  { rename } for  pid=1419 comm="init-xenstore-d" name="userdata-n.1.00000000-0000-0000-0000-000000000000.libxl-json" dev="sdb2" ino=4719263 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1
type=AVC msg=audit(1730402738.253:40): avc:  denied  { unlink } for  pid=1419 comm="init-xenstore-d" name="userdata-d.1.00000000-0000-0000-0000-000000000000.libxl-json" dev="sdb2" ino=4719273 scontext=system_u:system_r:xenstored_t:s0 tcontext=system_u:object_r:xend_var_lib_t:s0 tclass=file permissive=1

Signed-off-by: Cathy Hu <cahu@suse.de>

zpytela commented:
Merging, thank you.
